src/Microsoft.ApplicationInsights.AspNetCore/Extensions/ApplicationInsightsExtensions.cs

```diff
@@ -247,13 +247,54 @@ public static IServiceCollection AddApplicationInsightsTelemetryProcessor<T>(thi
 
             if (!telemetryProcessorType.GetTypeInfo().ImplementedInterfaces.Contains(typeof(ITelemetryProcessor)))
             {
-                throw new ArgumentException(nameof(telemetryProcessorType));
+                throw new ArgumentException(
+                    nameof(telemetryProcessorType) + " does not implement required ITelemetryProcessor interface.",
+                    nameof(telemetryProcessorType));
             }
 
             return services.AddSingleton<ITelemetryProcessorFactory>(serviceProvider =>
                 new TelemetryProcessorFactory(serviceProvider, telemetryProcessorType));
         }
 
+        /// <summary>
+        /// Adds the Application Insights telemetry initializer to the telemetry configuration.
+        /// </summary>
+        /// <typeparam name="T">Type of Telemetry Initializer to the add to the pipeline.</typeparam>
+        /// <param name="services">The <see cref="IServiceCollection"/> instance.</param>
+        /// <returns><see cref="IServiceCollection"/> instance.</returns>
+        public static IServiceCollection AddApplicationInsightsTelemetryInitializer<T>(this IServiceCollection services)
```

Comment:
Can you add a parameter constraint here? i.e. ` where T : ITelemetryInitializer`